src/dynamics/mpas/driver/dyn_mpas_subdriver.F90

```diff
@@ -1573,6 +2370,611 @@ subroutine dyn_mpas_exchange_halo(self, field_name)
+                                                                   edgenormalvectors(2, i) * north(2, cell2)  + &
+                                                                   edgenormalvectors(3, i) * north(3, cell2))
+        end do
+
+        nullify(nedges)
+
+        nullify(ucellzonal)
+        nullify(ucellmeridional)
+
+        nullify(cellsonedge)
+        nullify(east)
+        nullify(edgenormalvectors)
+        nullify(north)
+
+        nullify(uedge)
+
+        ! Make sure halo layers are up-to-date after computation.
+        call self % exchange_halo('u')
+
+        call self % debug_print(subname // ' completed')
+    end subroutine dyn_mpas_compute_edge_wind
+
+    !-------------------------------------------------------------------------------
+    ! subroutine dyn_mpas_init_phase4
+    !
+    !> \brief  Tracks `atm_core_init` to finish MPAS dynamical core initialization
+    !> \author Michael Duda
+    !> \date   29 February 2020
+    !> \details
+    !>  This subroutine completes MPAS dynamical core initialization.
+    !>  Essentially, it closely follows what is done in `atm_core_init`, but without
+    !>  any calls to MPAS diagnostics manager or MPAS stream manager.
+    !> \addenda
+    !>  Ported and refactored for CAM-SIMA. (KCW, 2024-05-25)
+    !
+    !-------------------------------------------------------------------------------
+    subroutine dyn_mpas_init_phase4(self, coupling_time_interval)
+        class(mpas_dynamical_core_type), intent(inout) :: self
+        integer, intent(in) :: coupling_time_interval ! Set the time interval, in seconds, over which MPAS dynamical core
+                                                      ! should integrate each time it is called to run.
+
+        character(*), parameter :: subname = 'dyn_mpas_subdriver::dyn_mpas_init_phase4'
+        character(strkind) :: date_time
+        character(strkind), pointer :: initial_time_1_pointer => null(), initial_time_2_pointer => null()
+        character(strkind), pointer :: xtime_pointer => null()
+        integer :: ierr
+        integer, pointer :: nvertlevels => null(), maxedges => null(), maxedges2 => null(), num_scalars => null()
+        logical, pointer :: config_do_restart => null()
+        real(rkind), pointer :: config_dt => null()
+        type(field0dreal), pointer :: field_0d_real => null()
+        type(mpas_pool_type), pointer :: mpas_pool => null()
+        type(mpas_time_type) :: mpas_time
+
+        call self % debug_print(subname // ' entered')
+
+        if (real(coupling_time_interval, rkind) < 1.0_rkind) then
+            call self % model_error('Invalid coupling time interval ' // stringify([real(coupling_time_interval, rkind)]), &
+                subname, __LINE__)
+        end if
+
+        call self % get_variable_pointer(config_dt, 'cfg', 'config_dt')
+
+        if (config_dt < 1.0_rkind) then
```

Comment:
I wouldn't say that a sub-one second timestep is necessarily an invalid timestep for the MPAS-A dycore. Admittedly it's unlikely that we'll be using such small timesteps any time soon in CAM-SIMA. But, it does seem like a somewhat arbitrary choice to disallow, say, a 0.5 s timestep, but to allow a 1.5 s timestep.
Perhaps instead we could simply check for negative values of `config_dt`?

Reply:
I pushed https://github.com/ESCOMP/CAM-SIMA/pull/267/commits/6428560b42e2eee6e8db89d1ffe04f1016212cd9 to relax this check. I think zero should also be invalid because it would make the time integration stall.